### PR TITLE
useOnEsc hook

### PR DESCRIPTION
**Pull request by PiotrKozlowski**
for: https://app.zenhub.com/workspaces/openstax-unified-5b71aabe3815ff014b102258/issues/openstax/unified/814

Those tests are tricky and need more work.

### CODE REVIEW

File: src/app/content/highlights/components/Confirmation.tsx [hunk 1]
```diff
@@ -46,8 +48,13 @@ interface Props {
 
 // tslint:disable-next-line:variable-name
 const Confirmation = ({message, confirmMessage, confirmLink, always, onCancel, onConfirm, ...props}: Props) => {
+  const element = React.useRef<HTMLElement>(null);
+
+  useOnEsc(element, true, onCancel);
 
   return <Overlay
+    ref={element}
+    focus={true}
```
Comment (TomWoodward) [inline src/app/content/highlights/components/Confirmation.tsx]: this overlay is not being focused when it is added, i'm not sure this prop does anythign

File: src/app/content/highlights/components/Card.tsx [hunk 2]
```diff
@@ -58,6 +58,8 @@ const Card = (props: Props) => {
   const [editing, setEditing] = React.useState<boolean>(!annotation);
   const locationFilters = useSelector(selectHighlights.highlightLocationFilters);
 
+  useOnEsc(element, props.isFocused, props.blur);
```
Comment (TomWoodward) [inline src/app/content/highlights/components/Card.tsx]: this isn't working when `DisplayNote` is rendered

**Review by TomWoodward — CHANGES_REQUESTED** [review]
this approach seems to be working very well. it seems like we have a few places where the focus element is not being managed perfectly though.

**Comment by TomWoodward** [comment]
@PiotrKozlowski i think it would be reasonable to avoid the places which are requiring extra work for now and address those in future pull requests. this is working well in the places where the focus element is set correctly

File: src/app/content/highlights/components/HighlightsPopUp.tsx [hunk 3]
```diff
@@ -131,6 +135,20 @@ class HighlightsPopUp extends Component<Props> {
     const popUp = this.popUp.current;
     if (isHtmlElement(popUp)) {
       popUp.focus();
+
+      if (this.removeEvListener) {
+        this.removeEvListener();
+      }
+
+      const [addEvListener, removeEvListener] = onEsc(popUp, this.props.closeMyHighlights);
```
[the remark this answers is not in the document]
Comment (PiotrKozlowski) [reply src/app/content/highlights/components/HighlightsPopUp.tsx]: yes

**Comment by TomWoodward** [comment]
@PiotrKozlowski checks are failing, code looks good thogh

**Comment by TomWoodward** [comment]
@PiotrKozlowski linter errors
```
src/app/utils.spec.tsx(3,1): error TS6133: 'renderer' is declared but its value is never read.
src/app/utils.spec.tsx(6,1): error TS6133: 'renderToDom' is declared but its value is never read.
```

**Comment by PiotrKozlowski** [comment]
@TomWoodward Thanks, it looks like `yarn lint:typescript`didn't catch this

**Review by TomWoodward — APPROVED** [review]
the new hook could be moved to the new reactUtils file, but i'm passing to qa now

**Comment by TomWoodward** [comment]
@PiotrKozlowski i tried this and didn't like the behavior so i did the forwardRef thing you mentioned in https://github.com/openstax/rex-web/pull/496#discussion_r378847619

i think this is good now

**Comment by PiotrKozlowski** [comment]
@TomWoodward Nice, thanks

**Pull request merged**